### PR TITLE
Makes nautical directions a proc

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request
Adds two new procs: dir2nautical and nautical2dir for converting between integer directions and their nautical text equivalent.
## Why It's Good For The Game
Deduplication of code. Instead of having these functions in each event that uses it, they can just call the proc instead. It will be used by more than one event now that we have directional code.
## Changelog
:cl: LT3
refactor: Nautical direction conversion gets its own set of procs
/:cl:

### CODE REVIEW

File: code/__HELPERS/type2type.dm [hunk 1]
```diff
@@ -39,6 +39,34 @@
...
+	switch(uppertext(direction))
+		if("FORE")
+			return NORTH
+		if("AFT")
+			return SOUTH
+		if("STARBOARD")
+			return EAST
+		if("PORT")
+			return WEST
+		else
```
Comment (ZephyrTFA) [inline code/__HELPERS/type2type.dm:67]: else... what?

Comment (lessthnthree) [reply code/__HELPERS/type2type.dm:67]: I was following the other procs above it. Rather it have an else return null, or nothing at all, then?

Comment (ZephyrTFA) [reply code/__HELPERS/type2type.dm:67]: It should CRASH

Comment (lessthnthree) [reply code/__HELPERS/type2type.dm:91]: Done.

**Review by ZephyrTFA — APPROVED** [review]

File: code/__HELPERS/type2type.dm [hunk 2]
```diff
@@ -59,7 +61,35 @@
...
+/proc/nautical2dir(direction)
+	switch(uppertext(direction))
+		if("FORE")
+			return NORTH
+		if("AFT")
+			return SOUTH
+		if("STARBOARD")
+			return EAST
+		if("PORT")
+			return WEST
```
Comment (san7890) [inline code/__HELPERS/type2type.dm:90]: would be cool if FORE/AFT/STARBOARD/PORT were defines like the cardinal directions "are"

Comment (lessthnthree) [reply code/__HELPERS/type2type.dm:90]: That can be done!

**Comment by Kapu1178** [comment]
What if someone makes a map that isnt aligned north

**Comment by Iamgoofball** [comment]
i hate nautical directions

**Comment by lessthnthree** [comment]
> What if someone makes a map that isnt aligned north

N/S/E/W are built into BYOND.

**Comment by ZephyrTFA** [comment]
> > What if someone makes a map that isnt aligned north
> 
> N/S/E/W are built into BYOND.

I believe what they meant by this, is that if the front of the "ship" is facing lets say west, all of these directions are rotated 90deg CCW

**Comment by san7890** [comment]
it doesn't fundamentally make sense from a mapping standpoint to say "this map is actually facing west", it is simpler from:

The Mapping Standpoint
A Player's Standpoint
A Code Standpoint (directionals)

to assume that North is always "Up". I believe it is a really bad idea to arbitrarily say "we're oriented aft-wise, but aft is up". Pretty much everything over the last 12 years was predicated on the fact that north is up, and I think it is just too injurious to allow a map that just decides to fuck with areas/directions in such a confusing manner.

We should also probably just get rid of direction-based names for areas entirely but that's something for another day

**Comment by lessthnthree** [comment]
> it doesn't fundamentally make sense from a mapping standpoint to say "this map is actually facing west", it is simpler from:
> 
> The Mapping Standpoint A Player's Standpoint A Code Standpoint (directionals)
> 
> to assume that North is always "Up". I believe it is a really bad idea to arbitrarily say "we're oriented starboard-wise, but starboard is up". Pretty much everything over the last 12 years was predicated on the fact that north is up, and I think it is just too injurious to allow a map that just decides to fuck with areas/directions in such a confusing manner. maintainability is a large concern here if it takes an averagely skilled mapper who wants to update your map that level of confusion where they think "oh, this guy musta screwed up their nautical directions!" and then "fixes" it.
> 
> We should also probably just get rid of direction-based names for areas entirely but that's something for another day

Yeah, and we're a station, not a ship. Buildings use N/S/E/W. I would be in favor of changing the nautical directions on the maps to cardinal, but I'd want to know there's acceptance of the idea before working on it.

**Comment by san7890** [comment]
> Yeah, and we're a station, not a ship. Buildings use N/S/E/W. I would be in favor of changing the nautical directions on the maps to cardinal, but I'd want to know there's acceptance of the idea before working on it.

@EOBGames and I have already signed off on the concept of making IceBox (a planetary map) to use Cardinal Directions, someone just has to do it.

The thing is that it only makes sense on IceBox because it's safe to assume that it has some level of magnetism that allows for these cardinal directions to be trustable- not so much on the other four non-planetary stations unfortunately.

**Comment by lessthnthree** [comment]
> > Yeah, and we're a station, not a ship. Buildings use N/S/E/W. I would be in favor of changing the nautical directions on the maps to cardinal, but I'd want to know there's acceptance of the idea before working on it.
> 
> @EOBGames and I have already signed off on the concept of making IceBox (a planetary map) to use Cardinal Directions, someone just has to do it.
> 
> The thing is that it only makes sense on IceBox because it's safe to assume that it has some level of magnetism that allows for these cardinal directions to be trustable- not so much on the other four non-planetary stations unfortunately.
> 
> This PR is a good first step for allowing this to happen since we can probably generate "name abstraction" hooking it into areas, so we can re-use the same definition to mean both "fore" and "north"... though I would honestly prefer making it two seperate area typepaths so they can have two distinct icons in a map editor.

Okay, I'll start looking at Icebox itself as the first step.

**Comment by lessthnthree** [comment]
Do this after feature freeze.

**Pull request closed without merging**